src/api/config/personalAttributionsConfig.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+
+import {
+  type IdentityId,
+  type Name,
+  identityIdParser,
+  nameParser,
+} from "../../core/identity";
+import {type Ledger} from "../../core/ledger/ledger";
+import {
+  type TimestampISO,
+  timestampISOParser,
+  fromISO,
+} from "../../util/timestamp";
+import * as C from "../../util/combo";
+import {type PersonalAttributions} from "../../core/credrank/personalAttribution";
+
+/**
+  This type resembles the JSON schema for configuring personal attributions,
+  which allows participants to attribute their cred to other participants.
+  This feature should not be used to make cred sellable/transferable, but
+  instead is intended to allow participants to acknowledge that a portion of
+  their creditted outputs are directly generated/supported by the labor of
+  others. (e.g. when a contributor has a personal assistant working behind the 
+  scenes)
+ */
+export type PersonalAttributionsConfig = Array<{|
+  // The name of a participant that is flowing cred to others
+  fromParticipantName: Name,
+  // The id of the participant. Will be autopopulated if absent.
+  fromParticipantId?: IdentityId,
```

Comment:
I understand that keying on the participant name is tenuous, but I think exposing both in a config should be handled more rigorously. If both are present, both should be validated, or the config should error as invalid if an account merge doesn't point towards a clear update for both values. In the case that the `IdentityId` was merged into another Identity, the name needs to be updated as well, since it's a mandatory config.

if we are going to update config values automatically, we should avoid letting the name go stale, basically.

Reply:
The name is updated automatically already. See line 65. And because it can go stale and merges/renames don't store records of previously names, keying on name after the ID has been added to the config isn't feasible. I would say we could remove the name after replacing it with ID, but then edits are harder for human users.